systemtest/src/test/java/io/strimzi/systemtest/performance/TopicOperatorPerformance.java

```diff
@@ -737,6 +741,112 @@ void testCapacity(String maxBatchSize, String maxBatchLingerMs) throws IOExcepti
         }
     }
 
+    private static Stream<Arguments> provideConfigurationsForTestSystemScalability() {
+        final int seed = 50;
+        final int limit = 20; // up to 1000 topics to create
+
+        // this means we would invoke create/update/delete operations x times in one iteration
+        int[] batchEventSizes = IntStream.iterate(seed, n -> n + seed).limit(limit).toArray();
+        return Stream.of(
+            new Object[][]{
+                    {"100", "100"},     // Default configuration
+                    {"100", "10"}       // Default batch size, with lower linger time
+                }
+            )
+            .flatMap(config -> Arrays.stream(batchEventSizes)
+                .mapToObj(numberOfTopics -> {
+                    int eventPerTask = 3;
+                    int numberOfTasks = numberOfTopics / eventPerTask;
+                    int spareEvents = numberOfTopics % eventPerTask;
+
+                    return Arguments.of(config[0], config[1], numberOfTasks, spareEvents);
+                }));
+    }
```

Comment:
That's a good idea. IMO, the full test is something to run on demand, mainly to check some big change to the Topic Operator. This is something every maintainer or contributor can do while developing or reviewing, not just me :)

Reply:
Yeah well, I agree that this should do every maintainer or contributor. But I know how it ends actually. I think it would be more user-friendly if these kind of tests could be more configurable. It's also better for debugging the issues, where you would just pass the config to the test and would not have to exclude the testcases or waiting a long time before you come to the point which interests you.

That way you will be able to remove the cool downs and other stuff that are present in the suite.

Reply:
I think cooling down is not bad, but we can reduce to 10 or 5 seconds.

Reply:
I agree with Lukas. Why do we have this pipeline that seems to be quite complex, AFAIU it is not easy to trigger just a specific configuration but you have to run all of the tests and wait...it doesn't sounds great. We have troubles for executing acceptance tests or regression on TF so having one huge pipeline with this execution time is not much usable from my POV. 

Why we don't have something a little bit shorter? Like scenario where you will create 500 topics and check it will be finished in some reasonable time? Or do we already have this kind of baseline that could be executed as part of regression? Kafka has it's benchmarks tests and AFAIR these doesn't take 10+hours.

Reply:
The objective here is to test the TO scalability, so you need to scale with increasing number of topic events. This test doesn't take that long when running locally, but I see your concerns when running on CI pipeline with less resources. In order to make it faster we can maybe run a shorter sequence of batches (i.e. 200, 400, 800, 1000). This takes around 2 minutes on my laptop. Would that work? I'm also not sure on which pipeline this would run and how it can be triggered.

Reply:
I guess even 1-2 hours are fine, but when I saw the pipeline running in Jenkins last week, it timeouted after 22h which is crazy. The timeout on TF is not set to 30h which is crazy as well. If it takes only 2 minutes, why we need 30h timeout? Regarding resources....30GB RAM and 8CPU is not so small I would say.

Reply:
> If it takes only 2 minutes, why we need 30h timeout?

It's around two minutes with the changes I proposed in my previous comment.

I actually didn't review the pipeline config as I don't know it very well, but I guess it was tuned for the initial implementation a needs to be revised now. 

@see-quick knows more for sure.

Reply:
In the previous commits, we have like 7 different configurations and each configuration tried to go from 50 to 1000 events (increasing per 50). Now we have only 2 configurations. :) So execution time is now not a problem. 

> As I mentioned offline, it's fine that you are doing these things, but what about having some smaller configuration that will be run for every pipeline and then some possibility to users that will run the tests (IMHO it will be mainly @fvaleri) to actually configure the stuff with what the tests should be run.
With this, it will run for a long time and in case that the test fails, you will, again, have to re-run all the tests? That doesn't apply just for this test, but for all other perf tests.

Okay, so we can decrease everything to pick only two configurations... 👍 as we have now this scalability test WDYT?